### DISCUSSION TITLE
Coming legislation and law that will likely affect the practice of greener software

### DISCUSSION THREAD

**Discussion by mrchrisadams** [Green-Software-Foundation/discussions#32] [Policy Working Group]
Hey folks, I've set up this post as a prompt for members to suggest examples of coming legisation or laws that are likely to have a a material impact on how easy it is to practice more environmentall sustainable software engineering practices.

Knowing they are coming makes it possible to come out with position statements, responses to consultations or briefings for decision makers to help inform the policy making process.

The recent [state of green software report insight](https://stateof.greensoftware.foundation/insights/software-legislation-has-quadrupled-in-the-last-decade/) has more on the recent rise in leglislation, and there are tools for seeing existing legislation and searching it, like the [climate policy radar](https://climatepolicyradar.org/), I am not aware of that many forward looking calendars or similar resources yet.

**Comment by mrchrisadams** [comment]
OK, I'll go first.

In Germany, the Energy Efficiency Law can be considered the precursor to the energy efficiency directive landing in the EU in May 2024. It's expected to pass in late Q3 early Q4, around September.

This is a good english language summary:
https://www.bundesregierung.de/breg-en/news/the-energy-efficiency-act-2184958

This is some good commentary:
https://www.datacenterdynamics.com/en/news/germany-to-pass-energy-efficiency-act-demanding-heat-reuse-in-data-centers

Here's a law firm Martin Dale (I think) giving some extra commentary too:
https://www.martindale.com/legal-news/article_greenberg-traurig-llp_2558886.htm

Given the struggles we've had with getting energy data for the SCI data and so on, this part is of interest:

> E. Information and advice in customer relations (Section 28 of the draft bill)
> 
> As of March 1, 2023, data center operators shall be required to transparently present the following to customers:
> 
> 1. the energy consumption per year directly attributable to customers, and
> 
> 2. the energy consumption of the technical infrastructure of the data center to be allocated in accordance with the consumption shares. 
> 

How this passes is likely to set a precedent for how the other 27 member states in the EU will need to implement the Energy Efficiency Directive in their own laws.

**Comment by mrchrisadams** [comment]
Related to the German Energy Efficiency Law, is the Europe-wide Energy Efficiency Directive being updated for targets set in 2030. 

You can see similarities between this and the German law - that's not an accident.

> *Minimum requirements for monitoring and publishing the energy performance of data centres*
> 
> The following minimum information shall be monitored and published as regards the energy performance of data centres referred to in Article 11(10):
> 
> (a)the name of the data centre; the name of the owner and operators of the data centre; the municipality where the data centre is based;
> 
> (b)the floor area of the data centre; the installed power; the annual incoming and outgoing data traffic; and the amount of data stored and processed within the data centre.
> 
> (c)the performance, during the last full calendar year, of the data centre in accordance with key performance indicators about, inter alia, energy consumption, power utilisation, temperature set points, waste heat utilisation, water usage and use of renewable energy.

Source: [Proposal for a DIRECTIVE OF THE EUROPEAN PARLIAMENT AND OF THE COUNCIL on energy efficiency (recast)](https://eur-lex.europa.eu/legal-content/EN/TXT/HTML/?uri=CELEX:52021PC0558)by

Green Software Foundation members [the SDIA have a good article on some more of the details](https://sdialliance.org/blog/new-energy-eu-efficiency-directive-targets-data-centers/). Having ~~access~~ _little or no access_ to power figures makes coming up with meaningful SCI scores much harder, so if the law mandated this kind of reporting it would likely make reporting or generating useful metrics much easier.

**Reply by ArneTR** [reply]
Hey Chris, thanks for putting the work in here in this post.

Quick question: Why does having access to power figures make SCI score reports harder? Shouldn't it be the other way round? Or did you mean "no access" ?

**Reply by NAMRATA-WOKE** [reply]
@mrchrisadams

**Reply by mrchrisadams** [reply]
DUH! yeah, oops. that should totally be the other way around. Lemme add it now.
